### PR TITLE
Queen Fixes And Buffs

### PR DESCRIPTION
**Pull request by Max-023**
## About The Pull Request

Gives the queen the defender's tail swipe and also reduces aging requirements.

## Why It's Good For The Game

Right now the queen is an awkward amalgamation of castes. Having abilities from the shrike, hivelord, and spitter, and the armor of a T3 frontline. As a support role, she is the peak with the exception of a lack of healing ability, but what she makes up for is her resin spit, construction, node planting, and egg laying. She can full construct a nest without issue. It is combat where the queen truly has issues.

Not having a true stun, the queen has a difficult time dealing with opponents eager to fight her. While resin spit can slow Marines down, this only helps her be able to maintain distance. Once the Marines get close, she has no form of punishment and will likely die if the other castes don't support her well. The queen screech, while technically a stun in playstyle, works for half a second and has a unjustifiably long cool down timer. It is not a reliable stun she can use in combat, resulting in her being severely exposed.

Additionally, she has an absurd time to age with very little benefits, the only notable thing being an increase in the leader cap. It is extremely hard for a queen to survive for long enough to reach ancient, requiring her to hide and leave her hive to do all the fighting just so she can get to elder empress. Not to mention both elder and ancient require 2000 ticks to pass individually, resulting in a whopping 4500 ticks for reaching ancient. You would have to start as the xeno queen immediately to even hope to see ancient, assuming the round is a long one.

## Changelog
:cl: Guprei
add: Adds the tail swipe ability to her abilities datums and her caste datums
balance: Brings down time to age dramatically
balance: Fixes armor to be consistent for each value
balance: Slight buff to speed
/:cl:

### CODE REVIEW

**Comment by psykzz** [comment]
+1 to giving queen tail sweep but not the balance changes.
Additionally make sure to clean up the tailsweep since its got some flags that dont apply to queen (crested for example)

**Comment by Pariah919** [comment]
why did you make queen evolve as literally fast as a t2

**Comment by Kuro-ai** [comment]
+1 for tail sweep,speed buffs are ehhh,but aging buffs are pretty insane,-1 to those

**Comment by Max-023** [comment]
The queen evolves slower than a T3, however. Their gains are relatively low compared to what the queen originally got, so age being super slow is unjustifiable if all you get is another leader and then some stat increases.

**Comment by MacBlaze1** [comment]
remove the balance changes and keep the tail sweep you balance illiterate spoon

**Comment by Max-023** [comment]
I have tweaked the aging to be only slightly slower than a T3 now.

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 1]
```diff
@@ -169,6 +169,77 @@
...
+			H.Paralyze(20)
+		GLOB.round_statistics.defender_tail_sweep_hits++
+		SSblackbox.record_feedback("tally", "round_statistics", 1, "tail_sweep_hits")
+		shake_camera(H, 2, 1)
+
+		to_chat(H, "<span class='xenowarning'>We are struck by \the [X]'s tail sweep!</span>")
+		playsound(H,'sound/weapons/alien_claw_block.ogg', 50, 1)
+
+	succeed_activate()
+
```
Comment (psykzz) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:233]:
```suggestion
	succeed_activate()
	add_cooldown()

```

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 2]
```diff
@@ -169,6 +169,78 @@
...
+			var/affecting = H.get_limb(ran_zone(null, 0))
+			if(!affecting) //Still nothing??
+				affecting = H.get_limb("chest") //Gotta have a torso?!
+			var/armor_block = H.run_armor_check(affecting, "melee")
+			H.apply_damage(damage, BRUTE, affecting, armor_block) //Crap base damage after armour...
+			H.apply_damage(damage, STAMINA) //...But some sweet armour ignoring Stamina
+			UPDATEHEALTH(H)
+			H.Paralyze(20)
+		GLOB.round_statistics.defender_tail_sweep_hits++
+		SSblackbox.record_feedback("tally", "round_statistics", 1, "tail_sweep_hits")
```
Comment (psykzz) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:226]: keep these in sync

**Comment by InterroLouis** [comment]
This causes Ancient Empress within an hour of roundstart so...can we maybe not?

**Comment by psykzz** [comment]
Is there really that big of a difference from young to ancient right now.

**Pull request merged**